contracts/walrus/sources/blob_manager.move

```diff
@@ -0,0 +1,833 @@
+/// Returns the blob's ObjectID if found, aborts otherwise.
+public fun get_blob_object_id_by_blob_id_and_deletable(
+    self: &BlobManager,
+    blob_id: u256,
+    deletable: bool,
+): ID {
+    let mut blob_info_opt = self.storage.find_blob(blob_id, deletable);
+    assert!(blob_info_opt.is_some(), EBlobNotRegisteredInBlobManager);
+    let blob_info = blob_info_opt.extract();
+    blob_info.object_id()
+}
+
+/// Returns the coin stash balances: (WAL, SUI).
+public fun coin_stash_balances(self: &BlobManager): (u64, u64) {
+    self.coin_stash.balances()
+}
+
+/// Returns the current extension policy.
+public fun extension_policy(self: &BlobManager): &ExtensionPolicy {
+    &self.extension_policy
+}
+
+/// Returns the grace period in epochs after storage expiry before blobs become eligible for GC.
+public fun grace_period_epochs(self: &BlobManager): u32 {
+    self.grace_period_epochs
+}
+
+// === Fund Manager Functions ===
+
+/// Withdraws all WAL funds from the coin stash.
+/// Requires fund_manager permission.
+public fun withdraw_all_wal(
+    self: &mut BlobManager,
+    cap: &BlobManagerCap,
+    ctx: &mut TxContext,
+): Coin<WAL> {
+    // Verify the capability.
+    check_cap(self, cap);
+    // Ensure the caller has fund_manager permission.
+    assert!(cap.fund_manager, ERequiresFundManager);
+
+    // Withdraw all WAL.
+    self.coin_stash.withdraw_all_wal(ctx)
+}
+
+/// Withdraws all SUI funds from the coin stash.
+/// Requires fund_manager permission.
+public fun withdraw_all_sui(
+    self: &mut BlobManager,
+    cap: &BlobManagerCap,
+    ctx: &mut TxContext,
+): Coin<SUI> {
+    // Verify the capability.
+    check_cap(self, cap);
+    // Ensure the caller has fund_manager permission.
+    assert!(cap.fund_manager, ERequiresFundManager);
+
+    // Withdraw all SUI.
+    self.coin_stash.withdraw_all_sui(ctx)
+}
+
+/// Withdraws a specific amount of WAL funds from the coin stash.
+/// Requires fund_manager permission.
+public fun withdraw_wal(
+    self: &mut BlobManager,
+    cap: &BlobManagerCap,
+    amount: u64,
+    ctx: &mut TxContext,
+): Coin<WAL> {
+    // Verify the capability.
+    check_cap(self, cap);
+    // Ensure the caller has fund_manager permission.
+    assert!(cap.fund_manager, ERequiresFundManager);
+
+    // Withdraw the specified amount of WAL.
+    self.coin_stash.withdraw_wal(amount, ctx)
+}
+
+/// Withdraws a specific amount of SUI funds from the coin stash.
+/// Requires fund_manager permission.
+public fun withdraw_sui(
+    self: &mut BlobManager,
+    cap: &BlobManagerCap,
+    amount: u64,
+    ctx: &mut TxContext,
+): Coin<SUI> {
+    // Verify the capability.
+    check_cap(self, cap);
+    // Ensure the caller has fund_manager permission.
+    assert!(cap.fund_manager, ERequiresFundManager);
+
+    // Withdraw the specified amount of SUI.
+    self.coin_stash.withdraw_sui(amount, ctx)
+}
+
+/// Sets the extension policy to fund_manager only.
```

Comment:
See suggestions above that might simplify the following 3 function, ie setting a default admin can always extend + just parameter.